### PR TITLE
ELK components for user-tracking analytics

### PR DESCRIPTION
**Pull request by lbjay**
Includes:

  * install/configure Elasticsearch, Logstash & Kibana
  * nginx reverse proxy for exposing kibana & ES Kopf plugin
  * Matterhorn user action harvester
  * refactor of ssl cert generation
  * automated elasticsearch snapshots to s3

### CODE REVIEW

File: files/default/elasticsearch-logrotate.conf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/var/log/elasticsearch/*.log {
+    daily
+    missingok
+    rotate 10
+    compress
+    notifempty
+    copytruncate
+}
```
Comment (djcp) [inline files/default/elasticsearch-logrotate.conf]: Nitpick - please add a newline to this file. Some things can be super picky about this.

File: recipes/configure-logstash-kibana.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  user 'kibana'
+  group 'kibana'
+  mode '644'
+  variables({
+    elasticsearch_host: es_host
+  })
+end
+
+execute 'service kibana restart'
+
```
Comment (djcp) [inline recipes/configure-logstash-kibana.rb]: super weird - trailing spaces but no newline? Can you make your editor add newlines always? vim and emacs do this, every editor should.  I know this is anal, but [it is the standard](https://en.wikipedia.org/wiki/Text_file#Unix_text_files).

File: CHANGELOG.md [hunk 3]
```diff
@@ -15,6 +15,10 @@
 
         ./bin/rake stack:commands:execute_recipes_on_layers layers="Admin, Workers, Engage" recipes="mh-opsworks-recipes::rsyslog-to-loggly"
 
+* *OPTIONAL EDITS TO THE CLUSTER CONFIG*:
+  Recipes, etc for setting up Analytics layer/node. See README.analytics.md in 
+  mh-opsworks for setup instructions.
```
Comment (djcp) [inline CHANGELOG.md]: We might want to mention re-running the other nginx recipes given that the refactor touched how the ssl certs are getting written.

**Comment by djcp** [comment]
Nothing incredibly major. :+1:

File: recipes/configure-logstash-kibana.rb (hunk 2 above)
Comment (lbjay) [inline recipes/configure-logstash-kibana.rb]: Yep. I've been using Aptana for this stuff but didn't take the time to set this stuff up. I'm switching to Atom and have enabled all the whitespace control stuff now.

File: CHANGELOG.md (hunk 3 above)
Comment (lbjay) [inline CHANGELOG.md]: Are you saying those recipes should be re-run even though nothing would have changed about the outcome of running them?

**Pull request closed without merging**